syntax = "proto3";

import "google/protobuf/empty.proto";

service Streamer {
  rpc GetCamProps(google.protobuf.Empty) returns (CamProps) {}
  rpc ReadFrames(ReadFramesRequest) returns (stream Frame) {}
}

message CamProps {
  int32 frame_width = 1;
  int32 frame_height = 2;
}

message ReadFramesRequest {
  int32 client_id = 1;
  string dtype = 2;
  int32 width = 3;
}

message Frame {
  bytes frame = 1;
}